docker_cli/_docker_cuda.py

```diff
@@ -0,0 +1,284 @@
+
+    @abstractmethod
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        """
+        Generates the first lines of the cuda install dockerfile.
+
+        These lines typically acquire the necessary CUDA repo and configure the
+        image to install CUDA.
+
+        Returns
+        -------
+        str
+            The initial lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @abstractmethod
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        """
+        Generates the install lines for the cuda install dockerfile.
+
+        These lines typically use the packet manager to install the given CUDA
+        libraries and then clean the packet manager cache.
+
+        Returns
+        -------
+        str
+            The install lines of the CUDA install dockerfile.
+        """
+        ...
+
+    @property
+    def package_manager(self) -> PackageManager:
+        """
+        The package manager used by this dockerfile generator.
+
+        This ensures that all subclasses of this class will have a package manager
+        property, but doesn't require the abstract base class to specify it.
+
+        Raises
+        ------
+        AttributeError
+            If no package manager is present on the object.
+        """
+        if not hasattr(self, "_package_manager"):
+            raise AttributeError(  # pragma: no cover
+                "Attribute 'package_manager' not found."
+            )
+        return getattr(self, "_package_manager")
+
+
+class AptGetCUDADockerfileGen(CUDADockerfileGenerator):
+    """
+    A CUDA dockerfile generator for debian-based apt-get systems
+    """
+
+    _dev_build_targets = [
+        "cuda-cudart-dev-$CUDA_PKG_VERSION",
+        "cuda-nvcc-$CUDA_PKG_VERSION",
+        "libcufft-dev-$CUDA_PKG_VERSION",
+    ]
+
+    def __init__(self, url_reader: URLReader):
+        super(AptGetCUDADockerfileGen, self).__init__(url_reader=url_reader)
+        self._package_manager: PackageManager = get_package_manager("apt-get")
+
+    def _generate_initial_lines(self, repo_ver: str, *, arch: str = "x86_64") -> str:
+        cuda_repo_name = (
+            "https://developer.download.nvidia.com/"
+            f"compute/cuda/repos/{repo_ver}/{arch}/"
+        )
+        filename = f"cuda-keyring_1.0-1_all.{self.package_manager.file_type}"
+        read_section = self.url_reader.generate_read_command(
+            target=f"{cuda_repo_name}" + filename,
+            output_file=filename,
+        )
+        package_command_section = self.package_manager.generate_package_command(
+            target=filename,
+        )
+        return f"RUN {read_section} \\\n && {package_command_section}"
+
+    def generate_install_lines(self, build_targets: Iterable[str]) -> str:
+        targets = build_targets
+        install_section = self.package_manager.generate_install_command(
+            targets=targets, clean=True
```

Comment:
Seems sort of pointless to rename `build_targets` to `targets` here in the function body

```suggestion
        install_section = self.package_manager.generate_install_command(
            targets=build_targets, clean=True
```